pytorch_lightning/utilities/cli.py

```diff
@@ -404,21 +404,25 @@ def __init__(
     def setup(self, trainer: Trainer, pl_module: LightningModule, stage: Optional[str] = None) -> None:
         # save the config in `setup` because (1) we want it to save regardless of the trainer function run
         # and we want to save before processes are spawned
-        log_dir = trainer.log_dir
+        log_dir = trainer.log_dir  # this broadcasts the directory
         assert log_dir is not None
+        fs = get_filesystem(log_dir)
         config_path = os.path.join(log_dir, self.config_filename)
-        if not self.overwrite and os.path.isfile(config_path):
-            raise RuntimeError(
-                f"{self.__class__.__name__} expected {config_path} to NOT exist. Aborting to avoid overwriting"
-                " results of a previous run. You can delete the previous config file,"
-                " set `LightningCLI(save_config_callback=None)` to disable config saving,"
-                " or set `LightningCLI(save_config_overwrite=True)` to overwrite the config file."
-            )
+        if not self.overwrite:
+            if fs.isfile(config_path):
+                raise RuntimeError(
```

Comment:
> I guess so that all processes terminate when this happens.

Yes

> if it needs to raise it also triggers the termination of the other processes?

This would also work. We'd need to broadcast the boolean decision.

Reply:
Done